src/adaptors/http1/http1_client.c

```diff
@@ -1238,6 +1505,24 @@ uint64_t qdr_http1_client_core_link_deliver(qdr_http1_adaptor_t    *adaptor,
     _client_response_msg_t *rmsg = DEQ_TAIL(hreq->responses);
     assert(rmsg && rmsg->dlv == delivery);
 
+    // when aggregating responses, they are saved on the list until
+    // the request has been settled, then encoded in the configured
+    // aggregation format
+    if (hconn->cfg.aggregation != QD_AGGREGATION_NONE) {
+        qd_log(qdr_http1_adaptor->log, QD_LOG_DEBUG, "[C%"PRIu64"][L%"PRIu64"] Received response (%i responses received), settling", hconn->conn_id, link->identity, DEQ_SIZE(hreq->responses));
+        rmsg->dispo = PN_ACCEPTED;
+        qd_message_set_send_complete(msg);
```

Comment:
Should this be predicated on qd_message_receive_complete(msg)?  I assume responses may be fragmented and we want to avoid updating the outcome until it is done arriving.

Reply:
Probably. I have added that check.